### PR TITLE
feat(icp): add redirections on http requests

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: common/src/http.rs [hunk 1]
```diff
@@ -21,14 +21,46 @@ mod icp {
...
+
+            match http_status_from_nat(&resp.status)? {
+                // moved_permamently / found / see_other / temporary_redirect / permament_redirect
+                // redirect and retry
+                301..=303 | 307..=308 => {
+                    url = resp
+                        .headers
+                        .iter()
+                        .find_map(|header| {
+                            if header.name == "location" || header.name == "Location" {
```
Comment (oblique) [inline common/src/http.rs]: The HTTP standard says that [field names are case-insensitive](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2). So use [eq_ignore_ascii_case](https://doc.rust-lang.org/stable/std/primitive.str.html#method.eq_ignore_ascii_case).

File: common/src/http.rs [hunk 2]
```diff
@@ -21,14 +21,46 @@ mod icp {
...
+                                Some(header.value.clone())
+                            } else {
+                                None
+                            }
+                        })
+                        .ok_or(HttpError::Other(format!(
```
Comment (oblique) [inline common/src/http.rs]: Use `ok_or_else` to avoid allocation on success

File: common/src/http.rs [hunk 3]
```diff
@@ -21,14 +21,46 @@ mod icp {
...
-        let req = CanisterHttpRequestArgument {
-            url: url.to_owned(),
-            method: HttpMethod::GET,
-            ..Default::default()
-        };
-        let resp = http_request(req, DEFAULT_HTTP_OUTCALL_COST).await?;
-        resp.0.try_into()
+        let mut url = url.to_owned();
+
+        for _ in 0..1 + MAX_REDIRECTS {
```
Comment (oblique) [inline common/src/http.rs]:
```suggestion
        for _ in 0..=MAX_REDIRECTS {
```

File: common/src/http.rs (hunk 1 above)
Comment (zvolin) [reply common/src/http.rs]: cool, made it exactly to not allocate on to_lowercase, wasn't aware of that method

**Review by oblique — APPROVED** [review]
lgtm

**Pull request merged**